Reuse JAVA SDK test suite (#39)
eclipse-esmf/esmf-sdk-py-aspect-model-loader: core/esmf-aspect-meta-model-python/README.md

    * [Download SAMM files](#download-samm-files)
      * [Download SAMM release](#download-samm-release)
      * [Download SAMM branch](#download-samm-branch)
    * [Download SAMM CLI](#download-samm-cli)
    * [Download test models](#download-test-models)
  * [Aspect Meta Model Loader usage](#aspect-meta-model-loader-usage)
  * [Samm Units](#samm-units)
  * [SAMM CLI wrapper](#samm-cli-wrapper)
* [Scripts](#scripts)
* [Automation Tasks](#automation-tasks)
  * [tox](#tox)
  * [GitHub actions](#github-actions)
    * [Check New Pull Request](#check-new-pull-request)
    * [Build release](#build-release)
<!-- TOC -->

# An Aspect of the Meta Model

The `esmf-aspect-model-loader` package provides the Python implementation for the SAMM Aspect Meta Model, or SAMM.
Each Meta Model element and each Characteristic class is represented by an interface with a corresponding
implementation.

## Set Up SAMM Aspect Meta Model

Before getting started to use the `esmf-aspect-model-loader` library you need to apply some set up actions:

Required
- [Install poetry](#install-poetry)
- [Install project dependencies](#install-project-dependencies)
- [Download SAMM files](#download-samm-files)
Optional

> - [Download SAMM CLI](#download-samm-cli) (need to use SAMM CLI functions)

Reviewer: ```suggestion
- [Download SAMM CLI](#download-samm-cli) (needed to use SAMM CLI functions)
```

Author: Fixed

- [Download SAMM test models](#download-test-models) (for running integration tests)

### Install poetry

`Poetry` used as a dependency management for the `esmf-aspect-model-loader`. Follow the next [instruction](https://python-poetry.org/docs/#installation)
 to install it.

To check the poetry version run:
```console
poetry --version
```

### Install project dependencies

Poetry provides convenient functionality for working with dependencies in the project. 
To automatically download and install all the necessary libraries, just run one command
```console
poetry install
```

### Download SAMM files

> There are two possibilities to download the SAMM files and extract the Turtle sources for the Meta Model: 

Reviewer: I think this section should also explain for whom are the SAMM files interesting, or in other words, why would you want to download them. IMO users of the aspect-meta-model-python should not have to deal with this at all (if they don't want to), including that they should not need to know about this.

SAMM release or SAMM branch

#### Download SAMM release

This script downloads a release JAR-file from GitHub, extracts them for further usage in the Aspect Model Loader:

To run script, execute the next command.
```console
poetry run download-samm-release
```
The version of the SAMM release is specified in the python script.

Link to all Releases: [SAMM Releases](https://github.com/eclipse-esmf/esmf-semantic-aspect-meta-model/releases)

#### Download SAMM branch

The script uses the GitHub API and downloads the files from the `main` GitHub branch. 

If the script is run in a pipeline, it uses a GitHub token to authorize. If the script is run locally, 
the API is called without a token. This may cause problems because unauthorized API calls are limited.

Run the next command to download and start working with the Aspect Model Loader.
```console
poetry run download-samm-branch
```
Link to all branches: [SAMM Releases](https://github.com/eclipse-esmf/esmf-semantic-aspect-meta-model/branches)

> ### Download SAMM CLI

Reviewer: This section I don't understand, shouldn't this work transparently? I.e., users should not be required to know about this at all, it should "just work". samm-cli should be downloaded automatically "under the hood" when it's not present yet.

The SAMM CLI is a command line tool for the validation of Aspect Models and the generation of artifacts, 
such as documentation or code, from Aspect Models.

Run the next command to download the SAMM client:
```console
poetry run download-samm-cli
```

> How to work with SAMM CLI from the Aspect Model Loader you can read in the chapter [SMM ClI wrapper](#samm-cli-wrapper).

Reviewer: ```suggestion
How to work with SAMM CLI from the Aspect Model Loader you can read in the chapter [SAMM CLI wrapper](#samm-cli-wrapper).
```

Author: Fixed

More detailed information about SAMM CLI functionality can be found in the 
[SAMM CLI documentation](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html).

> ### Download test models

Reviewer: IMO this also does not belong in the README. Shouldn't this be done automatically when the tests are run?

# ...
# symbol: V
```

## SAMM CLI wrapper

The SAMM CLI is a command line tool provided number of functions for working with Aspect Models.

> ore detailed information about SAMM CLI functionality can be found in the 

Reviewer: This paragraph is missing the start of a sentence

Author: Fixed

[SAMM CLI documentation](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html).

> Python aspect Model Loader provide a wrapper class to be able to call SAMM CLI functions from the python code.

Reviewer: ```suggestion
Python Aspect Model Loader provide a wrapper class to be able to call SAMM CLI functions from the Python code.
```

Author: Fixed

> For instance, validation a model can be done with the next code snippet

Reviewer: ```suggestion
For instance, validation of a model can be done with the following code snippet:
```

Author: Fixed

```python
from esmf_aspect_meta_model_python.samm_cli_functions import SammCli

samm_cli = SammCli()

> model_path = "Paht_to_the_model//Model.ttl"

Reviewer: ```suggestion
model_path = "Path_to_the_model/Model.ttl"
```

Author: Fixed
